cpp/src_prims/metrics/batched/silhouette_score.cuh

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "../silhouette_score.cuh"
+
+#include <common/device_buffer.hpp>
+
+#include <thrust/device_vector.h>
+#include <raft/cuda_utils.cuh>
+#include <raft/device_atomics.cuh>
+#include <rmm/device_uvector.hpp>
+#include <rmm/exec_policy.hpp>
+
+namespace MLCommon {
+namespace Metrics {
+namespace Batched {
+
+namespace detail {
+
+template <typename value_t, typename value_idx, typename label_idx>
+__global__ void fill_b_kernel(value_t *b, label_idx *y, value_idx n_rows,
```

Comment:
Don't need to doc all the arguments, but a quick doc w/ a description of these kernels would be helpful